recipes/libxml2/all/conanfile.py

```diff
@@ -67,29 +92,16 @@ def requirements(self):
             self.requires("icu/68.2")
 
     def build_requirements(self):
-        if self.settings.compiler != "Visual Studio" and tools.os_info.is_windows and os.environ.get("CONAN_BASH_PATH", None) is None:
-            self.build_requires("msys2/20190524")
-
-    @property
-    def _is_msvc(self):
-        return self.settings.compiler == 'Visual Studio'
-    
-    @property
-    def _is_mingw(self):
-        return self.settings.compiler == 'gcc' and self.settings.os == 'Windows'
+        if not self._is_msvc:
+            if self.options.zlib or self.options.lzma or self.options.icu:
+                self.build_requires("pkgconf/1.7.3")
+            if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```

Comment:
msys2 is only needed for mingw or am I seeing this wrong? I have often mentioned that this change makes cross compiling on windows for android no longer possible and you install msys2 unnecessarily when using windows clang for example.

Reply:
This logic was already there (and all recipes where autotools is used for mingw@windows) before this PR:
1. all triplets will run configure except if Visual Studio compiler
2. you need a shell to run configure
3. there is no shell on windows by default
4. 1+2+3 => we need msys2 if build machine is windows and compiler is not Visual Studio

The reason why it fails for cross compiling doesn't come from this logic, but from conan or msys2 recipe which shouldn't fail in msys2 `configure()` method when you cross compile from windows.

A workaround is to set `CONAN_BASH_PATH`.

Reply:
> This logic was already there
I don't know why cscript/configure.js is limited to VS only on Windows. Actually msys2 is not needed here at all, not even for MinGW.

Reply:
I know, it was removed in https://github.com/conan-io/conan-center-index/pull/2149. I would like to revert this modification but maybe not in this PR since it requires some effort.